public/services/shared-link.ts

```diff
@@ -103,14 +103,20 @@ export function setClipboardAndTarget(
   newValue: string,
   originalValue: string
 ) {
-  const range = document.createRange() as any;
-  const referenceNode = document.getElementsByTagName('span').item(0);
+  let range = document.createRange() as any;
+  const referenceNode = target;
+
+  const selection = document.getSelection();
+  if (selection?.rangeCount === 1) {
+    range = selection.getRangeAt(0);
```

Comment:
not a blocker, but thoughts on adding a test for this block?

Reply:
The tests I added in the [FTR repo](https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1128) already cover this case since there is only a single Selection once the Copy Link button is selected.

Its not possible for a selection to have multiple ranges in chrome because a user must select contiguous text.

Reply:
According to MDN Docs: https://developer.mozilla.org/en-US/docs/Web/API/Selection/addRange

> Note: Currently only Firefox supports multiple selection ranges, other browsers will not add new ranges to the selection if it already contains one.